frontend/pages/TagView.js

```diff
@@ -1,14 +1,181 @@
+    const fortyPctMax = Math.round(0.4 * maxNumOfPhotos);
+    const sixtyPctMax = Math.round(0.6 * maxNumOfPhotos);
+    const eightyPctMax = Math.round(0.8 * maxNumOfPhotos);
+    return (<>
+        {
+            sortedMapData.map((mapSquareData) => {
+                const index = mapSquareData.number;
+                const coords = mapSquareData.topLeftCoords;
+                const numberOfPhotos = mapSquareData.num_photos;
+
+                const mapSquareBounds = [
+                    [coords.lat, coords.lng],
+                    [coords.lat - MAPSQUARE_HEIGHT, coords.lng - MAPSQUARE_WIDTH]
+                ];
+                const link = "/map_square/" + index;
+                let mapSquareBucket = "";
+                // set of conditionals to calculate photo density for heat map
+                if (numberOfPhotos > 0 && numberOfPhotos <= twentyPctMax) {
+                    mapSquareBucket = "map-square box-one";
+                } else if (numberOfPhotos <= fortyPctMax) {
+                    mapSquareBucket = "map-square box-two";
+                } else if (numberOfPhotos <= sixtyPctMax) {
+                    mapSquareBucket = "map-square box-three";
+                } else if (numberOfPhotos <= eightyPctMax) {
+                    mapSquareBucket = "map-square box-four";
+                } else if (numberOfPhotos <= maxNumOfPhotos) {
+                    mapSquareBucket = "map-square box-five";
+                }
+
+                return (
+                    <Rectangle
+                        className={numberOfPhotos === 0 ? "map-grid" : mapSquareBucket}
+                        key={index}
+                        bounds={mapSquareBounds}>
+                        {// Shows map square numbers on each map square
+                            // <Marker position={L.polygon(mapSquareBounds).getBounds().getCenter()}
+                            //icon={L.divIcon({html: index})} />
+                        }
+                        <Popup>
+                            Map Square {index} <br/>
+                            <a href={link}>{numberOfPhotos} photos to show</a>
+                        </Popup>
+                    </Rectangle>
+                );
+            })
+        }
+    </>);
+}
+
+function arrondissementsOverlay(data) {
+    return data !== null ? data.map(tract => {
+        return (
+            <GeoJSON
+                style={{
+                    fillColor: "none",
+                    color: "#20CCD7"
+                }}
+                key={tract.properties["GISJOIN"]}
+                data={tract}
+            />
+        );
+    }) : <></>;
+}
 
 class TagView extends PhotoViewer {
     constructor(props) {
         super(props);
+
+        this.state = {
+            mapData: null,
+            geojsonData: null,
+            filledMapSquares: null,
+            filledMapSquaresData: null,
+            isLgViewportUp: null,
+            mapSquare: null,
+            mapLat: 48.858859,
```

Comment:
it looks like these coordinates are used across the project, maybe it's worth adding a constant in the `ParisMap` component to use instead of copying/pasting every time.